### PR TITLE
utils: add fmt::formatter for utils types

### PR DESCRIPTION
**Pull request by tchaikov**
before this change, we rely on the default-generated fmt::formatter created
from operator<<, but fmt v10 dropped the default-generated formatter.

in this change, we define formatters for 

* utils::human_readable_value
* std::strong_ordering
* std::weak_ordering
* std::partial_ordering
* utils::exception_container

Refs https://github.com/scylladb/scylladb/issues/13245

### CODE REVIEW

File: utils/to_string.cc [hunk 1]
```diff
@@ -10,39 +10,54 @@
 
 namespace std {
 
-std::ostream& operator<<(std::ostream& os, const std::strong_ordering& order) {
+std::ostream& boost_test_print_type(std::ostream& os, const std::strong_ordering& order) {
```
Comment (xemul) [inline utils/to_string.cc:13]: This is for testing, isn't it? In that case, it's better to localize it in test/ directory

Comment (tchaikov) [reply utils/to_string.cc]: yes, they are. i am moving them into `test/lib/test_utils.{hh,cc}`. does this make more sense now?

**Comment by tchaikov** [comment]
v2:

- move `boost_test_print_type()` overloads for `std::strong_ordering` and friends into `test/lib/test_utils.{hh,cc}`

File: utils/to_string.cc (hunk 1 above)
Comment (xemul) [reply utils/to_string.cc]: Yes, it's much better now, thank you.

Now I have out-of-curiosity question (not a merge blocked) -- why do we need those helpers at all? Boost uses operator<< to print types and doesn't "see" the newly introduced formatters, right?

Comment (tchaikov) [reply utils/to_string.cc]: yeah, Boost.test does not use {fmt} or even std::format, it uses either operator<< and falls back to  `boost_test_print_type()`, if the former is not available. actually, i didn't have to go through the fuss of renaming `operator<<` to `boost_test_print_type()` and moving them to their new home. but i really want to make it explicit that these helpers are for testing, and not supposed to be a part of our formatting system.

**Review by denesb — APPROVED** [review]

**Pull request merged**